packages/synthetix-main/contracts/modules/SUSDTokenModule.sol

```diff
@@ -0,0 +1,74 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@synthetixio/core-contracts/contracts/ownership/OwnableMixin.sol";
+import "@synthetixio/core-contracts/contracts/proxy/UUPSProxy.sol";
+import "@synthetixio/core-contracts/contracts/initializable/InitializableMixin.sol";
+import "@synthetixio/core-contracts/contracts/satellite/SatelliteFactory.sol";
+import "../interfaces/ISUSDTokenModule.sol";
+import "../storage/SUSDTokenStorage.sol";
+import "../token/SUSDToken.sol";
+import "../mixins/SUSDMixin.sol";
+
+contract SUSDTokenModule is
+    ISUSDTokenModule,
+    SUSDTokenStorage,
+    SUSDMixin,
+    OwnableMixin,
+    InitializableMixin,
+    SatelliteFactory
+{
+    event SUSDTokenCreated(address snxAddress);
+
+    function _isInitialized() internal view override returns (bool) {
+        return _sUSDInitialized();
+    }
+
+    function isSUSDTokenModuleInitialized() external view override returns (bool) {
+        return _isInitialized();
+    }
+
+    function initializeSUSDTokenModule() external override onlyOwner onlyIfNotInitialized {
+        SUSDTokenStore storage store = _sUSDTokenStore();
+
+        SUSDToken firstSUSDTokenImplementation = new SUSDToken();
+
+        UUPSProxy sUSDTokenProxy = new UUPSProxy(address(firstSUSDTokenImplementation));
+
+        address sUSDTokenProxyAddress = address(sUSDTokenProxy);
+        SUSDToken sUSDToken = SUSDToken(sUSDTokenProxyAddress);
+
+        sUSDToken.nominateNewOwner(address(this));
+        sUSDToken.acceptOwnership();
+        sUSDToken.setNewAuthorized(address(this));
+        sUSDToken.initialize("SyntUSD Token", "sUSD", 18);
```

Comment:
typo

Reply:
also I the official title should probably be something like "Synthetic USD Token v3"

Reply:
Fixed